src/shared/loop-util.c

```diff
@@ -688,34 +693,46 @@ int loop_device_make_by_path(
         } else if (open_flags < 0)
                 open_flags = O_RDWR;
 
+        /* Open the file also without O_DIRECT for unaligned probing */
```

Comment:
Thanks, I somehow read manual page of `fcntl()` wrong, so I thought that it's not possible to toggle O_DIRECT. Then there's probably no need to have two fds, since O_DIRECT could be simply enabled after probing (or disabled during probing) using only one fd.